jsx/Modal.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * This file contains the React Component for a Modal Window.
+ *
+ * @author Henri Rabalais
+ * @version 1.1.0
+ *
+ */
+import swal from 'sweetalert2';
+
+/**
+ * Modal Component.
+ * React wrapper for a Modal Window. Allows to dynamically toggle a Modal window.
+ *
+ * ================================================
+ * Usage:
+ * - Wrap the contents to be displayed by the Modal Window by the Modal Component.
+ * - Use the 'show' prop to toggle the Modal Component's presentation.
+ * - Use the 'title' prop to set a title for the Modal Component.
+ * - Use the 'onSubmit' prop set a submission function for the Modal's contents.
+ * - Use the 'throwWarning' prop to throw a warning upon closure of the Modal Window
+ * =================================================
+ *
+ */
+class Modal extends React.Component {
+  constructor() {
+    super();
+    this.closeModal = this.closeModal.bind(this);
+  }
+
+  closeModal() {
+    if (this.props.throwWarning) {
+      swal({
+        title: "Are You Sure?",
+        text: "Leaving the form will result in the loss of any information entered.",
+        type: "warning",
+        showCancelButton: true,
+        confirmButtonText: 'Proceed',
+        cancelButtonText: 'Cancel',
+      }).then(result => {
+        if (result.value) {
+          this.props.closeModal();
+        }
+      });
+    } else {
+      this.props.closeModal();
+    }
+  }
+
+  render() {
+    const headerStyle = {
+      display: 'flex',
+      flexDirection: 'row',
+      alignItems: 'center',
+      height: '40px',
+      borderTopRightRadius: '10',
+      fontSize: 24,
+      padding: 35,
+      borderBottom: '1px solid #DDDDDD'
+    };
+
+    const glyphStyle = {
+      marginLeft: 'auto',
+      cursor: 'pointer'
+    };
+
+    const bodyStyle = {
+      padding: 15
+    };
+
+    const children = () => this.props.show && this.props.children;
```

Comment:
Which part? The arrow function or the conditional allocation of `this.props.children`?

Reply:
I suspect he means the "conditional allocation". We should be able to generally assume that people looking at javascript can figure out an arrow function (moreso as time moves on and they become more standard.)

P.S. What does this line do?

Reply:
using `&&` short circuiting to conditionally return righthand value. just a matter of preference tho

Reply:
So it's basically `const children = this.props.children || this.props.show`?

Reply:
@davidblader I've sort of gotten used to this shorthand because it's really useful for quick checks that don't necessitate a ternary operator. Much like `||`, the null coalesce equivalent of `??` in php, the `&&` for setting values or triggering functions uses the same boolean-check principle and has become quite common practice from what I've seen. Not only that, but the principle isn't javascript specific and works in php too! 

@driusan when the function `children` is called, it will check whether `this.props.show` is true. If so, it will return `this.props.children`, else it will return `null`.